packages/@aws-cdk/aws-apigatewayv2/lib/common/base.ts

```diff
@@ -45,7 +50,10 @@ export abstract class StageBase extends Resource implements IStage {
    * @internal
    */
   protected _addDomainMapping(domainMapping: DomainMappingOptions) {
-    new ApiMapping(this, `${domainMapping.domainName}${domainMapping.mappingKey}`, {
+    if (this.apiMapping) {
+      throw new Error('_addDomainMapping cannot be called more than once for a Stage');
```

Comment:
```suggestion
      throw new Error('Only one ApiMapping allowed per Stage');
```

Reply:
Add a test case for this.

Reply:
@nija-at What is the recommended pattern for testing a protected function?

Reply:
I see the problem, it might be a bit convoluted.
I think it's fine to skip the test case for this. Sorry for the mixed message.

Reply:
Not a problem! Thanks for the response